mp/src/game/shared/momentum/mom_system_gamemode.cpp

```diff
@@ -423,23 +430,23 @@ IGameMode *CGameModeSystem::GetGameMode(int eMode) const
     return m_vecGameModes[eMode];
 }
 
-IGameMode *CGameModeSystem::GetGameModeFromMapName(const char *pMapName)
+IGameMode *CGameModeSystem::GetGameModeFromString(const char *pInput)
 {
-    if (!pMapName || !pMapName[0])
-        return nullptr;
+    if (!pInput || !pInput[0])
+        return m_vecGameModes[GAMEMODE_UNKNOWN];
 
     // Skip over unknown in the loop
     for (auto i = 1; i < m_vecGameModes.Count(); ++i)
     {
         const auto pPrefix = m_vecGameModes[i]->GetMapPrefix();
-        const auto strLen = Q_strlen(pPrefix);
-        if (!Q_strnicmp(pPrefix, pMapName, strLen))
+        const auto strLen = Q_strlen(pPrefix) - 1;
```

Comment:
Why the `- 1`?

Reply:
I'm assuming most players won't include the underscore when doing the cvar manually, and doing `- 1` removes it from the comparison to allow them to do so

Reply:
I've included a comment saying "- 1 to remove underscore from prefix, as most players wont include it" with the initialization to help any later confusion